src/main/java/io/odpf/firehose/factory/FirehoseConsumerFactory.java

```diff
@@ -80,14 +83,29 @@ public FirehoseConsumerFactory(KafkaConsumerConfig kafkaConsumerConfig, StatsDRe
         parser = new KeyOrMessageParser(new ProtoParser(stencilClient, kafkaConsumerConfig.getInputSchemaProtoClass()), kafkaConsumerConfig);
     }
 
+    private Filter buildFilter(FilterConfig filterConfig) {
+        instrumentation.logInfo("Filter Engine: ", filterConfig.getFilterEngine());
+        switch (filterConfig.getFilterEngine()) {
+            case JSON:
+                Instrumentation jsonFilterUtilInstrumentation = new Instrumentation(statsDReporter, JsonFilterUtil.class);
+                JsonFilterUtil.logConfigs(filterConfig, jsonFilterUtilInstrumentation);
+                JsonFilterUtil.validateConfigs(filterConfig, jsonFilterUtilInstrumentation);
+                return new JsonFilter(filterConfig, new Instrumentation(statsDReporter, JsonFilter.class));
+            case JEXL:
+                return new JexlFilter(filterConfig, new Instrumentation(statsDReporter, JexlFilter.class));
+            default:
+                throw new IllegalArgumentException("Invalid filter engine type");
```

Comment:
here we can return an implementation with NoOp which doesn't filter anything, then we won't need to rely on getFilterJsonDataSource() == NONE

Reply:
fixed